### PR TITLE
[-] Technical - Filter parser is now asynchronous

### PR DESCRIPTION
**Pull request by VincentMolinie**
## Why ?

This is a change that was needed in `forest-express-mongoose` and so it needed to change the `BaseFilterParser` to async. So this PR is too make the work easier for the next developer working on it (As it will be already asynchrone)

## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
   "dependencies": {
     "babel-runtime": "7.0.0-beta.3",
     "bluebird": "2.9.25",
-    "forest-express": "5.0.0",
+    "forest-express": "5.1.0",
```
Comment (rap2hpoutre) [inline package.json]: I think we should not have more than one subject in a PR (maybe you needed this to make your PR work), even if it's hard because of the `forest-express`/`forest-express-sequelize`/`forest-express-mongoose` dependencies.

According to @arnaudbesnier, changelog have to include all the changes that were added in `forest-express` when you update a dependency, so it would have complexified this actual PR (by adding several (unrelated) lines in readme and dependency file to review)

So I think it's hard to decorrelate both subject, but merging them makes it harder to understand, review, test and git-merge too.

Maybe I'm wrong, let me know if I missed something

File: src/services/resources-getter.js [hunk 2]
```diff
@@ -17,12 +17,14 @@ function ResourcesGetter(model, options, params) {
   const OPERATORS = new Operators(options);
   const primaryKey = _.keys(model.primaryKeys)[0];
   const filterParser = new FiltersParser(schema, params.timezone, options);
+  let fieldNamesRequested;
+  let searchBuilder;
```
Comment (rap2hpoutre) [inline src/services/resources-getter.js]: It seems there are more changes that just "Filter parser is now asynchronous" in this PR (+36/-24 in this `resources-getter` file). 

Could you also add a note in this ticket to explain why this file changed too? (it seems a function was renamed and its behavior as changed). Maybe you could update the changelog too? (or extract this part in one other PR if possible).

File: test/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  env: {
+    mocha: true,
+    node: true
+  },
+  rules: {
+    'no-unused-expressions': 0,
+    'arrow-body-style': 0
+  },
+  globals: {}
+}
```
Comment (rap2hpoutre) [inline test/.eslintrc.js]: I think this is a good addition :+1:. 

Though, this should be in another PR, to test it and release it separately. It could have been reviewed and accepted quickly in another PR (imagine if this one is not accepted, the eslint rules would not be accepted too, even if it's a good addition).

Comment (rap2hpoutre) [reply test/.eslintrc.js]: (or maybe I misunderstood something, but still, there could have been a comment on the PR)

File: package.json (hunk 1 above)
Comment (VincentMolinie) [reply package.json]: The update of this is required and is the only change actually. Between `forest-express` 5.0.0 and 5.1.0 the only thing changing is the parser being called asynchronously so The changelog is good for me as the PR won't work without that change

File: test/.eslintrc.js (hunk 3 above)
Comment (VincentMolinie) [reply test/.eslintrc.js]: As you requested: https://github.com/ForestAdmin/forest-express-sequelize/pull/332

Comment (VincentMolinie) [reply test/.eslintrc.js]: I'll rebase once validated

File: package.json (hunk 1 above)
Comment (rap2hpoutre) [reply package.json]: Thank you!

File: src/services/resources-getter.js (hunk 2 above)
Comment (rap2hpoutre) [reply src/services/resources-getter.js]: OK, thank you!
